### PR TITLE
feat(remap transform): accept more timestamp patterns in `to_timestamp`

### PR DESCRIPTION
**Pull request by JeanMertz**
Based on comment in https://github.com/timberio/vector/issues/3741#issuecomment-692075031.

### CODE REVIEW

File: src/mapping/query/functions.rs [hunk 1]
```diff
@@ -190,12 +190,14 @@ impl Function for ToTimestampFn {
...
 
     match value {
-        Value::Bytes(b) => DateTime::parse_from_rfc3339(&String::from_utf8_lossy(&b))
-            .map(|dt| dt.with_timezone(&Utc))
-            .map_err(|_| "cannot parse string as RFC3339".to_owned()),
+        Value::Bytes(b) => {
+            return Conversion::Timestamp
+                .convert(Value::Bytes(b))
+                .map_err(|e| e.to_string())
+        }
```
Comment (bruceg) [inline src/mapping/query/functions.rs:200]: Does this work?
```suggestion
        Value::Bytes(b) => Conversion::Timestamp
                .convert(value)
                .map_err(|e| e.to_string()),
```

**Review by bruceg — APPROVED** [review]

**Review by StephenWakely — APPROVED** [review]
I concur with Bruce's statement.

File: src/mapping/query/functions.rs (hunk 1 above)
Comment (JeanMertz) [reply src/mapping/query/functions.rs:200]: It doesn't, because `convert` returns a `Result<Value>`, whereas the other arms return `Result<DateTime>`.

Comment (bruceg) [reply src/mapping/query/functions.rs:200]: Oh, I see, I missed that in the reduced context of the diff.

Given the other two arms that return a value are pretty simple, I think removing the map would simplify things here.

**Pull request merged**